housekeeper/cli/tables.py

```diff
@@ -87,3 +89,67 @@ def get_versions_table(rows: List[dict]) -> Table:
             version_obj["expires_at"].split("T")[0] if version_obj["expires_at"] else "",
         )
     return table
+
+
+def squash_names(list_of_files):
+    """If subsequent elements (filenames) in 'list_of_files' end in an integer- And that integer is
+    following the previous- those are squashed when displayed.
+    Example:
+
+        ["asdf1.txt", "asdf2.txt"] becomes asdf[1-2].txt'
+    """
+    list_of_squashed = []
+    tag_list = []
+    if list_of_files == []:
+        return list_of_squashed
+    head, *tail = list_of_files
+    (previous_file, previous_counter, previous_suffix) = _get_suffix(head["path"])
+    squash = [previous_counter]
+    previous_hkjson = head
+    for hk_json in tail+[{'path':""}]:  # Add padding for final iteration
+        (filename, counter, suffix) = _get_suffix(hk_json["path"])
+        if counter == str(_to_int(previous_counter) + 1) and (previous_file == filename):
+            squash = squash+[counter]
+            tag_list = tag_list +(hk_json["tags"])
+        else:
+            if len(squash) == 1: # only previous element in list
+                list_of_squashed.append(previous_hkjson)
+                squash = [counter]
+            else:
+                squashed_path = previous_file+"["+squash[0]+"-"+squash[-1]+"]"+previous_suffix
+                previous_hkjson["path"]=squashed_path
+                previous_hkjson["tags"]= remove_duplicates(sorted(tag_list,key =lambda i:i["id"]))
+                previous_hkjson["id"]="-"
+                list_of_squashed.append(previous_hkjson)
+                squash = [counter]
+                tag_list = []
+        previous_counter = counter
+        previous_suffix = suffix
+        previous_file = filename
+        previous_hkjson = hk_json
+    return list_of_squashed
+
+
+def remove_duplicates(tag_list):
+    """Remove duoplicate elements from `tag_list`"""
+    no_duplicates = []
+    for i in tag_list:
+       if i not in no_duplicates:
+          no_duplicates.append(i)
+    return no_duplicates
```

Comment:
```suggestion
    # Remove all duplicated items
    return list(set(tag_list))
```
Would this work?

Reply:
No, I've tried.

But it is a list of dicts -which is not hashable. Set entries must be hashable. I believe the reason is that a hashable object must be immutable.